torchtune/modules/transforms/pipelines.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import logging
+from typing import Dict, Optional, Tuple, Union
+
+import numpy as np
+import PIL
+
+import torch
+
+from torchtune.modules.transforms.transforms import (
+    divide_to_equal_patches,
+    pad_image_top_left,
+    rescale,
+    ResizeWithoutDistortion,
+)
+from torchtune.modules.transforms.utils import GetBestResolution
+from torchvision.transforms.v2 import functional as F
+
+logger = logging.getLogger(__name__)
+
+ImageInput = Union["PIL.Image.Image", np.ndarray, "torch.Tensor"]
+
+
+class VariableImageSizeTransforms:
+    """
+    This class accepts images of any size and dynamically resize, pads and chunks it
+    based on the image aspect ratio and the number of image chunks we allow.
+
+    The algorithm will NOT distort the image fit a certain aspect ratio, because
```

Comment:
nit: this sentence reads weirdly - do you mean "The algorithm will NOT distort the image to fit a certain aspect ratio"?